### PR TITLE
Integrate sharded lvldb persister

### PR DESCRIPTION
**Pull request by ssd04**
## Reasoning behind the pull request
- Integrate sharded serial leveldb persister
- instantiate shard id persister for storage
  
## Proposed changes
- add config options for shard id persister
- update storage factory to include sharded persister

## Testing procedure
- TBD

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?

### CODE REVIEW

File: storage/disabled/shardIDProvider.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func (s *shardIDProvider) ComputeId(key []byte) uint32 {
+	return 0
+}
+
+// NumberOfShards returns 0
+func (s *shardIDProvider) NumberOfShards() uint32 {
+	return 0
+}
+
+// GetShardIDs return nil
```
Comment (BeniaminDrasovean) [inline storage/disabled/shardIDProvider.go:20]: returns

File: storage/disabled/shardIDProvider_test.go [hunk 2]
```diff
@@ -0,0 +1 @@
+package disabled
```
Comment (BeniaminDrasovean) [inline storage/disabled/shardIDProvider_test.go:1]: Do not forget about this

File: storage/pruning/fullHistoryPruningStorer_test.go [hunk 3]
```diff
@@ -295,14 +296,18 @@ func TestFullHistoryPruningStorer_ConcurrentOperations(t *testing.T) {
 
 	fmt.Println(testDir)
 	args := getDefaultArgs()
-	args.PersisterFactory = factory.NewPersisterFactory(config.DBConfig{
+	persisterFactory, err := factory.NewPersisterFactory(config.DBConfig{
```
Comment (BeniaminDrasovean) [inline storage/pruning/fullHistoryPruningStorer_test.go:299]: move `config.DBConfig{` on a new line

File: storage/pruning/pruningStorer_test.go [hunk 4]
```diff
@@ -1050,14 +1051,19 @@ func TestPruningStorer_ConcurrentOperations(t *testing.T) {
 
 	fmt.Println(testDir)
 	args := getDefaultArgs()
-	args.PersisterFactory = factory.NewPersisterFactory(config.DBConfig{
+
+	persisterFactory, err := factory.NewPersisterFactory(config.DBConfig{
```
Comment (BeniaminDrasovean) [inline storage/pruning/pruningStorer_test.go:1055]: move `config.DBConfig{` to a new line

File: storage/disabled/shardIDProvider.go (hunk 1 above)
Comment (ssd04) [reply storage/disabled/shardIDProvider.go:20]: done

File: storage/disabled/shardIDProvider_test.go (hunk 2 above)
Comment (ssd04) [reply storage/disabled/shardIDProvider_test.go:1]: right :+1: thanks

**Review by BeniaminDrasovean — APPROVED** [review]

**Review by AdoAdoAdo — APPROVED** [review]

File: storage/storageunit/constants.go [hunk 5]
```diff
@@ -16,6 +16,8 @@ const (
 	LvlDB = storageUnit.LvlDB
 	// LvlDBSerial represents a levelDB storage with serialized operations identifier
 	LvlDBSerial = storageUnit.LvlDBSerial
+	// ShardedLvlDBSerial represents a sharded levelDB storage with serialized operations identifier
```
[the remark this answers is not in the document]
Comment (ssd04) [reply storage/storageunit/constants.go:18]: updated comment

**Review by AdoAdoAdo — APPROVED** [review]

**Review by BeniaminDrasovean — APPROVED** [review]

**Pull request merged**